packages/app/src/cli/models/extensions/extension-instance.ts

```diff
@@ -335,11 +336,13 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
 
     if (this.features.includes('bundling')) {
       // Modules that are going to be inclued in the bundle should be built in the bundle directory
-      this.outputPath = joinPath(bundleDirectory, extensionId, outputFile)
+      this.outputPath = this.isFunctionExtension ? this.outputPath : joinPath(bundleDirectory, extensionId, outputFile)
```

Comment:
Can't modify outputpath which is the function's directory before compilation.

Reply:
Unfortunately the outputPath ends up pointing to a tmp directory so unless we are okay with not having the wasm file generated in the app's extension directory (which also might have unforeseen consequences, I haven't tried it) I think we need to avoid changing the outputPath if it is a function type extension.

Reply:
It looks like `outputPath` is used to know where the build artifact should be located so that it can be built/deployed/bundled/whatever. I think the problem is that this path is defined by the developer via the TOML file (since they control the build tooling), so we can't really change it ourselves.

One idea to work around this is that we could not change the `outputPath` for any extension. Then once the artifact is produced at that path, we move it into the bundle directory. This assumes that there's a single artifact produced for building an extension, and it's located at `outputPath`, but I'm not sure that's a safe assumption.

```diff
diff --git a/packages/app/src/cli/models/extensions/extension-instance.ts b/packages/app/src/cli/models/extensions/extension-instance.ts
index addb7eba72..928fd80039 100644
--- a/packages/app/src/cli/models/extensions/extension-instance.ts
+++ b/packages/app/src/cli/models/extensions/extension-instance.ts
@@ -334,15 +334,13 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
     const extensionId = identifiers?.extensions[this.localIdentifier] ?? this.configuration.uid ?? this.handle
     const outputFile = this.isThemeExtension ? '' : joinPath('dist', `${this.outputFileName}`)
 
-    if (this.features.includes('bundling')) {
-      // Modules that are going to be inclued in the bundle should be built in the bundle directory
-      this.outputPath = this.isFunctionExtension ? this.outputPath : joinPath(bundleDirectory, extensionId, outputFile)
-    }
-
     await this.build(options)
-    if (this.isFunctionExtension) {
-      await copyFile(this.outputPath, joinPath(bundleDirectory, extensionId, 'index.wasm'))
+
+    if (this.features.includes('bundling')) {
+      const bundleOutputPath = joinPath(bundleDirectory, extensionId, outputFile)
+      await moveFile(this.outputPath, bundleOutputPath)
     }
+
     if (this.isThemeExtension && useThemebundling()) {
       await bundleThemeExtension(this, options)
     }
~
~
```

Alternatively, if my assumption is not true, then we could have the `buildFunctionExtension` function move the file to where it needs to be. We'd need to pass in the final location of the file though (maybe through the `buildDirectory` option?). For example:

```diff
diff --git a/packages/app/src/cli/services/build/extension.ts b/packages/app/src/cli/services/build/extension.ts
index bf81e275fb..b980baa598 100644
--- a/packages/app/src/cli/services/build/extension.ts
+++ b/packages/app/src/cli/services/build/extension.ts
@@ -146,6 +146,9 @@ export async function buildFunctionExtension(
     } else {
       await buildOtherFunction(extension, options)
     }
+
+    const buildPath = joinPath(extension.directory, extension.config.build.path ?? joinPath('dist', 'index.wasm'))
+    await copyFile(buildPath, options.buildDirectory)
   } finally {
     await releaseLock()
   }
```

@isaacroldan WDYT?

Reply:
Works for me, thank you!